formula-android/src/main/java/com/instacart/formula/integration/FormulaFragmentViewFactory.kt

```diff
@@ -0,0 +1,34 @@
+package com.instacart.formula.integration
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.instacart.formula.android.FeatureView
+import com.instacart.formula.android.ViewFactory
+import com.instacart.formula.fragment.FragmentKey
+import java.lang.IllegalStateException
+
+internal class FormulaFragmentViewFactory(
+    private val key: FragmentKey,
+    private val featureProvider: FeatureProvider,
+) : ViewFactory<Any> {
+
+    private var factory: ViewFactory<Any>? = null
+
+    @Suppress("UNCHECKED_CAST")
+    override fun create(inflater: LayoutInflater, container: ViewGroup?): FeatureView<Any> {
+        val featureEvent = featureProvider.getFeature(key) ?: throw IllegalStateException("Missing feature for unknown reasons.")
```

Comment:
Maybe `Missing feature key` or `Could not find feature for $key.tag` would be easier to understand here.

Reply:
Good call, I will add `$key` to the message.